### PR TITLE
Chore: Migrated FIS, SSM, SQS, and STS to V2

### PR DESCRIPTION
**Pull request by edibble21**
Fixes #N/A 

**Description**
Migrated FIS, SSM, SQS, and STS to AWS SDK V2

**How was this change tested?**
make presubmit and make test

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/errors/errors.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ var (
 		"InvalidInstanceID.NotFound",
 		launchTemplateNameNotFoundCode,
 		"InvalidLaunchTemplateId.NotFound",
-		sqs.ErrCodeQueueDoesNotExist,
+		"QueueDoesNotExist",
```
Comment (engedaam) [inline pkg/errors/errors.go:41]: Is there a typed version of this error?

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by edibble21 — COMMENTED** [review]
/karpenter snapshot

File: test/suites/integration/tags_test.go [hunk 2]
```diff
@@ -106,10 +108,10 @@ var _ = Describe("Tags", func() {
 			env.ExpectCreated(nodeClass)
 			env.EventuallyExpectInstanceProfileExists(env.GetInstanceProfileName(nodeClass))
 
-			_, err := env.IAMAPI.UntagInstanceProfile(&iam.UntagInstanceProfileInput{
+			_, err := env.IAMAPI.UntagInstanceProfile(context.Background(), &iam.UntagInstanceProfileInput{
```
Comment (engedaam) [inline test/suites/integration/tags_test.go:111]:
```suggestion
			_, err := env.IAMAPI.UntagInstanceProfile(env.Context, &iam.UntagInstanceProfileInput{
```

File: test/suites/integration/tags_test.go [hunk 3]
```diff
@@ -118,11 +120,11 @@ var _ = Describe("Tags", func() {
 			env.EventuallyExpectKarpenterRestarted()
 
 			Eventually(func(g Gomega) {
-				out, err := env.IAMAPI.GetInstanceProfile(&iam.GetInstanceProfileInput{
+				out, err := env.IAMAPI.GetInstanceProfile(context.Background(), &iam.GetInstanceProfileInput{
```
Comment (engedaam) [inline test/suites/integration/tags_test.go:123]:
```suggestion
				out, err := env.IAMAPI.GetInstanceProfile(env.Context, &iam.GetInstanceProfileInput{
```

File: test/pkg/environment/aws/expectations.go [hunk 4]
```diff
@@ -422,13 +425,13 @@ func (env *Environment) ExpectInstanceProfileDeleted(instanceProfileName, roleNa
...
 	}
-	_, err := env.IAMAPI.RemoveRoleFromInstanceProfile(removeRoleFromInstanceProfile)
+	_, err := env.IAMAPI.RemoveRoleFromInstanceProfile(context.Background(), removeRoleFromInstanceProfile)
 	Expect(awserrors.IgnoreNotFound(err)).To(BeNil())
 
 	deleteInstanceProfile := &iam.DeleteInstanceProfileInput{
 		InstanceProfileName: aws.String(instanceProfileName),
 	}
-	_, err = env.IAMAPI.DeleteInstanceProfile(deleteInstanceProfile)
+	_, err = env.IAMAPI.DeleteInstanceProfile(context.Background(), deleteInstanceProfile)
```
Comment (engedaam) [inline test/pkg/environment/aws/expectations.go:434]: We need to make sure we are using the environment context 

```suggestion
	_, err = env.IAMAPI.DeleteInstanceProfile(env.Context, deleteInstanceProfile)
```

File: test/pkg/environment/aws/environment.go [hunk 5]
```diff
@@ -118,8 +123,9 @@ func NewEnvironment(t *testing.T) *Environment {
 	}
 	// Initialize the provider only if the INTERRUPTION_QUEUE environment variable is defined
 	if v, ok := os.LookupEnv("INTERRUPTION_QUEUE"); ok {
-		sqsapi := servicesqs.New(session)
-		out := lo.Must(sqsapi.GetQueueUrlWithContext(env.Context, &servicesqs.GetQueueUrlInput{QueueName: aws.String(v)}))
+		cfg := lo.Must(config.LoadDefaultConfig(context.Background(), config.WithRetryMaxAttempts(3)))
```
Comment (engedaam) [inline test/pkg/environment/aws/environment.go:126]:
```suggestion
		cfg := lo.Must(config.LoadDefaultConfig(env.Context, config.WithRetryMaxAttempts(3)))
```

File: pkg/controllers/controllers.go [hunk 6]
```diff
@@ -69,8 +71,9 @@ func NewControllers(ctx context.Context, mgr manager.Manager, sess *session.Sess
 		status.NewController[*v1.EC2NodeClass](kubeClient, mgr.GetEventRecorderFor("karpenter")),
 	}
 	if options.FromContext(ctx).InterruptionQueue != "" {
-		sqsapi := servicesqs.New(sess)
-		out := lo.Must(sqsapi.GetQueueUrlWithContext(ctx, &servicesqs.GetQueueUrlInput{QueueName: lo.ToPtr(options.FromContext(ctx).InterruptionQueue)}))
+		cfg := lo.Must(config.LoadDefaultConfig(context.Background(), config.WithRetryMaxAttempts(3)))
```
Comment (engedaam) [inline pkg/controllers/controllers.go:74]:
```suggestion
		cfg := lo.Must(config.LoadDefaultConfig(ctx, config.WithRetryMaxAttempts(3)))
```

File: pkg/operator/operator.go [hunk 7]
```diff
@@ -153,7 +153,7 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 		*sess.Config.Region,
 	)
 	versionProvider := version.NewDefaultProvider(operator.KubernetesInterface, cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval))
-	ssmProvider := ssmp.NewDefaultProvider(ssm.New(sess), cache.New(awscache.SSMGetParametersByPathTTL, awscache.DefaultCleanupInterval))
+	ssmProvider := ssmp.NewDefaultProvider(ssm.NewFromConfig(cfg), cache.New(awscache.SSMGetParametersByPathTTL, awscache.DefaultCleanupInterval))
 	amiProvider := amifamily.NewDefaultProvider(operator.Clock, versionProvider, ssmProvider, ec2api, cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval))
 	amiResolver := amifamily.NewDefaultResolver()
 	launchTemplateProvider := launchtemplate.NewDefaultProvider(
```
Comment (engedaam) [inline pkg/operator/operator.go:159]: We should include the EKS API change in this PR, no?

**Pull request closed without merging**